phpBB/phpbb/db/doctrine/connection_factory.php

```diff
@@ -97,14 +96,63 @@ public static function get_connection_from_params(
 
 		try
 		{
-			return DriverManager::getConnection($params);
+			$conn = DriverManager::getConnection($params);
+			Type::addType(case_insensitive_string::CASE_INSENSITIVE_STRING, case_insensitive_string::class);
+			$conn->getDatabasePlatform()->registerDoctrineTypeMapping('varchar_ci', case_insensitive_string::CASE_INSENSITIVE_STRING);
+			return $conn;
 		}
 		catch (Exception $e)
 		{
 			throw new runtime_exception('DB_CONNECTION_FAILED');
 		}
 	}
 
+	/**
+	 * Converts phpBB driver names to Doctrine's equivalent.
+	 *
+	 * @param string $driver_name phpBB database driver name.
+	 *
+	 * @return string Doctrine DBAL's driver name.
+	 *
+	 * @throws InvalidArgumentException If $driver_name is not a valid phpBB database driver.
+	 */
+	private static function to_doctrine_driver(string $driver_name) : string
+	{
+		// Normalize driver name.
+		$name = str_replace('phpbb\db\driver', '', $driver_name);
+		$name = preg_replace('/mysql$/i', 'mysqli', $name);
+		$name = trim($name, '\\');
+
+		switch ($name)
+		{
+			case 'mssql_odbc':
+			case 'mssqlnative':
+				$name = 'pdo_sqlsrv';
+				break;
```

Comment:
Break should be aligned with case
```suggestion
			break;
```